ikarus/finiteelements/mechanics/materials/hyperelastic/deviatoric/blatzko.hh

```diff
@@ -0,0 +1,132 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+/**
+ * \file BlatzKo.hh
+ * \brief Implementation of the BlatzKo material model.
+ * \ingroup  materials
+ */
+
+#pragma once
+
+#include <ikarus/finiteelements/mechanics/materials/interface.hh>
+#include <ikarus/finiteelements/mechanics/materials/materialhelpers.hh>
+#include <ikarus/finiteelements/mechanics/materials/tags.hh>
+#include <ikarus/utils/tensorutils.hh>
+
+namespace Ikarus::Materials {
+
+/**
+ * \brief Implementation of the Blatz-Ko material model.
+ * \ingroup materials
+ *
+ * \details The energy is computed as
+ * \f[ \hat{\Psi}(\la_1, \la_2, \la_3) = \frac{\mu}{2}(
+ \frac{1}{\la_1^2} + \frac{1}{\la_2^2} + \frac{1}{\la_3^2}
+ + 2\la_1 \la_2 \la_3 - 5). \f]
+ *
+ * \remark A special Blatz-Ko material model is implemented here. It assumes material parameters which corresponds to a
+ * Poisson's ratio of 0.25.
+ *
+ * \tparam ST The underlying scalar type.
+ */
+template <typename ST>
+struct BlatzKoT
+{
+  using ScalarType         = ST;
+  using PrincipalStretches = Eigen::Vector<ScalarType, 3>;
+
+  static constexpr int dim         = 3;
+  static constexpr auto stretchTag = PrincipalStretchTag::total;
+
+  using FirstDerivative  = Eigen::Vector<ScalarType, dim>;
+  using SecondDerivative = Eigen::Matrix<ScalarType, dim, dim>;
+
+  using MaterialParameters = double;
+
+  [[nodiscard]] constexpr static std::string name() noexcept { return "BlatzKo"; }
+
+  /**
+   * \brief Constructor for BlatzKoT.
+   * \param mu The shear modulus.
+   */
+  explicit BlatzKoT(MaterialParameters mu)
+      : mu_{mu} {}
+
+  /**
+   * \brief Returns the material parameters stored in the material
+   */
+  MaterialParameters materialParametersImpl() const { return mu_; }
+
+  /**
+   * \brief Computes the stored energy in the BlatzKo material model.
+   *
+   * \param lambda principal stretches
+   * \return ScalarType
+   */
+  ScalarType storedEnergyImpl(const PrincipalStretches& lambda) const {
+    return mu_ / 2 *
+           (1 / pow(lambda[0], 2) + 1 / pow(lambda[1], 2) + 1 / pow(lambda[2], 2) +
+            2 * lambda[0] * lambda[1] * lambda[2] - 5);
```

Comment:
This should be `lambda.prod()` https://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html#af119d9a4efe5a15cd83c1ccdf01b3a4f